pennylane/optimize/lie_gradient.py

```diff
@@ -0,0 +1,352 @@
+
+        return None
+
+    def get_su_n_operators(self):
+        r"""Get the 2x2 SU(N) operators. The dimension of the group is N^2-1.
+
+        Returns:
+            List of (N^2)x(N^2) numpy complex arrays and corresponding paulis words
+        """
+
+        operators = []
+        names = []
+        # get all possible combinations of paulis
+        pauli_combinations = it.product(
+            list(range(len(self.pl_paulis))), repeat=self.nqubits
+        )
+        # remove the identity
+        next(pauli_combinations)
+        # construct the corresponding pennylane observables
+        for ps in pauli_combinations:
+            operators.append(self.get_pauli_op(ps))
+            names.append(ps)
+        return operators, [
+            "".join([self.pauli_int_to_str[i] for i in n]) for n in names
+        ]
+
+    def get_pauli_op(self, paulistring):
+        r"""Create a qml.Observable for a String corresponding to a Pauli word.
+
+        Args:
+            paulistring: String of the form 'IXYI...Z' corresponding to a Pauli word
+
+        Returns:
+            qml.Observable
+        """
+        for idx, p in enumerate(paulistring):
+            if p != 0:
+                op = self.pl_paulis[p](idx)
+                for q in paulistring[idx + 1:]:
+                    idx += 1
+                    if q != 0:
+                        op @= self.pl_paulis[q](idx)
+                break
+        return op
+
+    def get_omegas(self):
+        r"""Measure the coefficients of the Lie gradient with respect to a Pauli word basis
+
+        We want to calculate the components of the Lie gradient with respect to a Pauli word basis
+        For a Hamiltonian of the form :math:`H = \sum_i O_i`, this can be achieved by calculating
+
+        .. math:
+
+            \omega_{i,j} = \text{Tr}([\rho, O_i] P_j)
+
+        where :math:`P_j` is a Pauli word in the set of Pauli monomials on :math:`N` qubits.
+
+        Via the parameter shift rule, the commutator can be calculated as
+
+        .. math:
+
+            [\rho, O_i] = \frac{1}{2}(V(\pi/2) \rho V^\dag(\pi/2) - V(-\pi/2) \rho V^\dag(-\pi/2))
+
+        where :math:`V` is the unitary generated by the Pauli word :math:`V(\theta) = \exp{-i\theta P_j }`.
+
+        Returns:
+            Array of omegas for each direction in the Lie algebra.
+
+        """
+
+        obs_groupings, _ = qml.grouping.group_observables(
+            self.observables, self.coeffs
+        )
+        # get all circuits we need to calculate the coefficients
+        circuits = self.algebra_commutator(self.circuit.qtape, obs_groupings)[0]
+        # For each observable O_i in the Hamiltonian, we have to calculate all Lie coefficients
+        omegas = np.zeros((len(self.coeffs), len(self.lie_algebra_basis_names)))
+        idx = 0
+        for circuit_plus, circuit_min in circuits:
+            out_plus = qml.execute(
+                circuit_plus, self.circuit.device, gradient_fn=qml.gradients.param_shift
```

Comment:
if you don't need these executions to be differentiable, you can set `gradient_fn=None` for a small speed boost :)